api/types.go

```diff
@@ -12,21 +15,31 @@ type ProxyConfig struct {
 }
 
 type OIDCConfig struct {
-	IssuerUrl    *string
-	ClientId     *string
-	ClientSecret *string
-	Audience     *string
+	IssuerUrl    *string `json:"issuer_url"`
+	ClientId     *string `json:"client_id"`
+	ClientSecret *string `json:"client_secret"`
+	Audience     *string `json:"audience"`
 	Enabled      *bool
+	Filename     *string
 }
 
 type TokenVerificationConfig struct {
 	Enabled *bool
 	Url     *string
 }
 
-func (c *OIDCConfig) Validate() {
+func (c *OIDCConfig) ReadAndValidate() {
 	if !*c.Enabled {
```

Comment:
Unrelated to this PR but worth mentioning anyway:
The if/else is not needed here.
If it is not enabled we return. Therefore the code after that in this method is only executed when it is enabled. This means that if/else is unnecessary.